### ISSUE TITLE
Named options or posargs?

### ISSUE THREAD

**Issue by msabramo** [msabramo/ansible_role_apply#1]
@sudarkoff said:

> I'd also replace named options with positional args. One of the things that irritate me about 
> ansible is all those named opts - too much typing! :)
> 
> E.g.: `ansible apply-role docker localhost`

One problem with posargs is it makes it harder to do multiple roles or multiple hosts, but maybe that’s ok.

**Comment by sudarkoff** [comment]
I see your point about multiple roles. But I feel like the 99% use case for this would be running a single role against a single host or a group of hosts.

**Comment by msabramo** [comment]
Yeah, I think I agree. Probably when you reach the point where you want to apply multiple roles, you're ready to be using playbooks.

**Comment by msabramo** [comment]
```
ansible-role-apply docker vagrant --sudo
Usage: ansible-role-apply [OPTIONS]

Error: Got unexpected extra arguments (docker vagrant)
```

**Comment by msabramo** [comment]
Another point. Using a named option for the role (i.e.: `-r/--role-name`) is more consistent with what the `ansible` command does for `-m/--module-name`. 

Though maybe that's different because the `ansible` command is more general and maybe we don't always want to follow ansible's lead and want to be different in purpose.

**Comment by msabramo** [comment]
Fixed by #3
